recipes/aws-sdk-cpp/all/conanfile.py

```diff
@@ -0,0 +1,260 @@
+            "ds",
+            "dynamodb",
+            "dynamodbstreams",
+            "ec2",
+            "ecr",
+            "ecs",
+            "eks",
+            "elasticache",
+            "elasticbeanstalk",
+            "elasticfilesystem",
+            "elasticloadbalancing",
+            "elasticloadbalancingv2",
+            "elasticmapreduce",
+            "elastictranscoder",
+            "email",
+            "es",
+            "events",
+            "firehose",
+            "fms",
+            "fsx",
+            "gamelift",
+            "glacier",
+            "globalaccelerator",
+            "glue",
+            "greengrass",
+            "guardduty",
+            "health",
+            "iam",
+            "identity-management",
+            "importexport",
+            "inspector",
+            "iot-data",
+            "iot-jobs-data",
+            "iot",
+            "kafka",
+            "kinesis",
+            "kinesisanalytics",
+            "kinesisvideo",
+            "kms",
+            "lambda",
+            "lex",
+            "lightsail",
+            "logs",
+            "machinelearnings",
+            "macie",
+            "marketplace-entitlement",
+            "marketplacecommerceanalytics",
+            "mediaconvert",
+            "medialive",
+            "mediapackage",
+            "mediastore",
+            "mediatailor",
+            "meteringmarketplace",
+            "mobileanalytics",
+            "monitoring",
+            "mq",
+            "mturk-requester",
+            "neptune",
+            "opsworks",
+            "opsworkscm",
+            "organizations",
+            "pinpoint",
+            "polly",
+            "pricing",
+            "queues",
+            "quicksight",
+            "ram",
+            "rds",
+            "redshift",
+            "recognition",
+            "resource-groups",
+            "robomaker"
+            "route53",
+            "route53domains",
+            "s3",
+            "sagemaker",
+            "sdb",
+            "serverlessrepo"
+            "servicecatalog",
+            "servicediscovery",
+            "shield",
+            "signer",
+            "sms",
+            "snowball",
+            "sns",
+            "sqs",
+            "ssm",
+            "states",
+            "storagegateway",
+            "sts",
+            "support",
+            "swf",
+            "text-to-speech",
+            "texttract",
+            "transcribe",
+            "transfer",
+            "translate",
+            "waf",
+            "workdocs",
+            "worklink",
+            "workmail",
+            "workspaces",
+            "xray"
+           )
+    options = {
+            **{ x: [True, False] for x in sdks},
+            **{
+                "shared": [True, False],
+                "fPIC": [True, False],
+                "min_size": [True, False],
+                }
+            }
+    default_options = {key: False for key in options.keys()}
```

Comment:
Two comments:

- the default upstream behavior is NOT core only by default, but all components (`BUILD_ONLY` empty: https://github.com/aws/aws-sdk-cpp/blob/2be13177875e944151132d90305dbc46e80bf8e3/cmake/sdks.cmake#L20-L70)!

- when you build all components, I think that `short_paths` is required. I see warnings like this one on Windows:

  ```
  CMake Warning in source_subfolder/aws-cpp-sdk-sagemaker-featurestore-runtime/CMakeLists.txt:
    The object file directory
  
      C:/Users/spaceim/.conan/data/aws-sdk-cpp/1.8.130/_/_/build/2222ca76f73f6ada7e5ba6a37e22584d8a0a2621/source_subfolder/aws-cpp-sdk-sagemaker-featurestore-runtime/CMakeFiles/aws-cpp-sdk-sagemaker-featurestore-runtime.dir/./
  
    has 220 characters.  The maximum full path to an object file is 250
    characters (see CMAKE_OBJECT_PATH_MAX).  Object file
  
      ub_SAGEMAKER-FEATURESTORE-RUNTIME.cpp.obj
  
    cannot be safely placed under this directory.  The build may not work
    correctly.
  ```

Reply:
I'll finish later because it's late here in France but thank you for these comments! I may have a few question but I'll ask them after a good night of sleep! 😉

Reply:
I've tried the build by default (with the 6 high level components enabled): it builds absolutely all the components (267 including core in 1.8.130).
I think that it's fine to leave only these 6 high level components as an option to reduce complexity in options.

Reply:
Yes that is why I started only with the core component. I will first add your new suggestions then implement the logic for the dependency (the module dependencies are explained here: https://github.com/aws/aws-sdk-cpp/blob/master/cmake/sdksCommon.cmake#L153-L158)

Where should I implement this logic? In the `config_options` method? Using `self.cpp_info.components["aws-sdk-cpp-foobar"].requires` ? Both?

In your suggestion you're changing the library names from `aws-cpp-sdk-foobar` to `aws-sdk-cpp-foobar` but are you aware that the SDK is not consistent on the library names? Prefix is almost always `aws-sdk-cpp-` except for generated libraries (ie: `aws-cpp-sdk-foobar.lib`)

Reply:
> Yes that is why I started only with the core component. I will first add your new suggestions then implement the logic for the dependency (the module dependencies are explained here: https://github.com/aws/aws-sdk-cpp/blob/master/cmake/sdksCommon.cmake#L153-L158)

Yes, it shows dependencies of high level components. I don't know if others components have also dependencies (except to core), maybe not.

> Where should I implement this logic? In the config_options method? Using self.cpp_info.components["aws-sdk-cpp-foobar"].requires ? Both?

First, in `package_info()`, these dependencies should appear in `requires` property of each component. So basically, each component depends on `core`. High level components also depend on several others components.
Then, in `configure()` or `package_id()`, options that can't be disabled when another is enabled should be deleted. For example when `queues` option is True, `sqs` option has no effect because `sqs` can't be disabled.

> In your suggestion you're changing the library names from aws-cpp-sdk-foobar to aws-sdk-cpp-foobar but are you aware that the SDK is not consistent on the library names? Prefix is almost always aws-sdk-cpp- except for generated libraries (ie: aws-cpp-sdk-foobar.lib)

I did not change libs names (only names for generators), or I don't remember and it was a mistake.